frontend/src/Pages/FrontScreen.js

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import { Button } from 'react-bootstrap';
+import Container from 'react-bootstrap/Container';
+import Row from 'react-bootstrap/Row';
+import { Link } from "react-router-dom";
+
+
+class FrontScreen extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      "auth": false
+    }
+  }
+
+  componentDidMount() {
+    this.authenticate();
+  }
+
+  authenticate() {
+    const microapps = window.microapps;
+    const request = { nonce: 'nonce typically generated server-side' };
+    microapps.getIdentity(request).then(response => {
+      const decoded = JSON.parse(atob(response.split('.')[1]));
+      this.setState(decoded);
```

Comment:
this looks wrong to me, there are two setState calls one after another, can you merge.